primitives/sr-io/src/lib.rs

```diff
@@ -92,16 +92,23 @@ pub trait Storage {
 	}
 
 	/// Returns the data for `key` in the child storage or `None` if the key can not be found.
-	fn child_get(&self, child_storage_key: &[u8], key: &[u8]) -> Option<Vec<u8>> {
+	/// A key collision free unique id is required as parameter.
+	fn child_get(
+		&self,
+		child_storage_key: &[u8],
+		child_definition: &[u8],
+		child_type: u32,
+		key: &[u8],
+	) -> Option<Vec<u8>> {
```

Comment:
I think, parameter here lacks documentation, also note that we can use any scale encodable structure here. But maybe its fine

Reply:
Yes that is one of the thing I did ask myself, is it worth separating both parameters.
My justification was that child info and child storage key should be handle separately because storage key can potentially change more often than child_definition (and is mostly use as key in our overlays and others temporary structure). But it sure could make sense to have both in the same field (especially if we consider a possibility of hierarchical trie where we would need to decode/encode the multiple level of storage key from/to child_storage_key depending on the child_type).
I will document that after rebasing to master.

Reply:
> My justification was that child info and child storage key should be handle separately because storage key can potentially change more often than child_definition (and is mostly use as key in our overlays and others temporary structure).

Yes I understand from the runtime perspective both are very linked and will be changed together, from the overlay (maybe trie level) having the key to uniquely identify a childtrie is enough.

I'm not sure of what is best either